ingestion/src/metadata/ingestion/source/metadata.py

```diff
@@ -161,10 +166,18 @@ def prepare(self):
     @classmethod
     def create(cls, config_dict, metadata_config: OpenMetadataConnection):
         config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
-        connection: MetadataESConnection = config.serviceConnection.__root__.config
-        if not isinstance(connection, MetadataESConnection):
+        connection = config.serviceConnection.__root__.config
+        if connection.type.value == "OpenMetadata" and not isinstance(
+            connection, OpenMetadataConnection
+        ):
+            raise InvalidSourceException(
+                f"Expected OpenMetadataConnection, but got {connection}"
+            )
+        elif connection.type.value == "MetadataES" and not isinstance(
+            connection, MetadataESConnection
```

Comment:
would it make sense to prepare different sources here? One for metadataES and one for openMetadata, as we treat other SQL sources. Otherwise it feels a bit weird to have a generic one where we need to pick up the specific class. WDYT?

Reply:
for different sources I mean source modules. E.g., openmetadata.py, metadata_es.py...

Reply:
Got it - thanks - I will ping once i am ready with the solution